alert-common/src/main/java/com/synopsys/integration/alert/common/security/authorization/AuthorizationManager.java

```diff
@@ -139,6 +139,23 @@ public final boolean hasAllPermissions(String context, String descriptorName, Ac
                    .anyMatch(name -> permissionCache.containsKey(name) && permissionCache.get(name).hasPermissions(permissionKey, operations));
     }
 
+    public Set<UserRoleModel> getRoles() {
+        return roleAccessor.getRoles();
+    }
+
+    public Set<UserRoleModel> getRoles(Collection<Long> roleIds) {
+        return roleAccessor.getRoles(roleIds);
+    }
+
+    public boolean doesRoleNameExist(String name) {
+        return permissionCache.containsKey(name);
```

Comment:
Now that the roles are being updated using methods in this class and the cache is being updated in those methods, is it alright to check the cache for the role name? Or should we still using the RoleAccessor to check the database?